### PR TITLE
saga and balance page

### PR DESCRIPTION
**Pull request by dzheki4ak**

### CODE REVIEW

File: src/components/BalancePage/BalancePage.jsx [hunk 1]
```diff
@@ -4,22 +4,42 @@ import { HOME } from '../../routes';
...
-      <span className="main__page_header-span"> 1034074028,00 💲</span>
-    </h3>
-    <h3 className="balance__page_received">
-      Total received:
-      <span className="main__page_header-span"> 1034074028,00 💲</span>
-    </h3>
-    <NavLink text={`Home`} route={HOME} />
-  </div>
-);
+const { useSelector } = require('react-redux');
```
Comment (vitcool) [inline src/components/BalancePage/BalancePage.jsx]: OK, as I see you have 
```
import NavLink from '../NavLinks/NavLink.jsx';
```
and here you use 
```
const { useSelector } = require('react-redux');
```
will be better to use common approach for importing 3d party dependencies

File: src/components/BalancePage/BalancePage.jsx [hunk 2]
```diff
@@ -4,22 +4,42 @@ import { HOME } from '../../routes';
...
+
+const BalancePage = () => {
+  const list = useSelector(state => state.transactionsList.transactionsList);
```
Comment (vitcool) [inline src/components/BalancePage/BalancePage.jsx]: I can advice move `state => state.transactionsList.transactionsList` to  `https://github.com/vitcool/testnet/blob/master/src/redux/modules/transactionsList/selectors.js` as
```
const getTransactionsList = (state) => state.transactionsList.transactionsList;
```
and use it here
```
const list = useSelector(getTransactionsList);
```

File: src/components/BalancePage/BalancePage.jsx [hunk 3]
```diff
@@ -4,22 +4,42 @@ import { HOME } from '../../routes';
...
+    .flat()
+    .map(input => input.output_value)
+    .reduce((inVal, inAcc) => inVal + inAcc);
+
+  const totalOutputs = list
+    .map(trns => trns.outputs)
+    .flat()
+    .map(output => output.value)
+    .reduce((outVal, outAcc) => outVal + outAcc);
+
```
Comment (vitcool) [inline src/components/BalancePage/BalancePage.jsx:20]: looks like calculating `totalInputs ` and `totalOutputs ` pretty much the same, could you move it to the helper function, please?

File: src/redux/modules/transactionsList/selectors.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+
+export const transactionsTotalsSelector = state =>
+  state.transactionsList.transactions;
+
+
```
Comment (vitcool) [inline src/redux/modules/transactionsList/selectors.js]: why do you have a lot of redundant empty lines here?)

File: src/redux/modules/transactionsList/selectors.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    .reduce((inVal, inAcc) => inVal + inAcc, 0) / 1000;
+
+export const sumOfallOutputsSelector = state =>
+  transactionsTotalsSelector(state)
+    .map(trns => trns.outputs)
+    .flat()
+    .map(output => output.value)
+    .reduce((outVal, outAcc) => outVal + outAcc, 0) / 1000;
+
+
```
Comment (vitcool) [inline src/redux/modules/transactionsList/selectors.js]: calculation `sumOfallInputsSelector ` and `sumOfallOutputsSelector ` seems the same, do you have any ideas show you can helper function and then reuse it in both scenarios?

Comment (vitcool) [reply src/redux/modules/transactionsList/selectors.js]: why name is `sumOfallInputsSelector` not `sumOfAllInputsSelector `?

File: src/redux/store.js [hunk 6]
```diff
@@ -1,9 +1,17 @@
 import { createStore, combineReducers, compose, applyMiddleware } from 'redux';
-import thunk from 'redux-thunk';
```
Comment (vitcool) [inline src/redux/store.js:2]: do you use `'redux-thunk'` anywhere else?
probably you need to remove it from the `package.json` also?

**Review by vitcool — COMMENTED** [review]
could you check my comments, please?

File: src/components/BalancePage/BalancePage.jsx (hunk 1 above)
Comment (dzheki4ak) [reply src/components/BalancePage/BalancePage.jsx]: Fixed

File: src/redux/modules/transactionsList/selectors.js (hunk 4 above)
Comment (dzheki4ak) [reply src/redux/modules/transactionsList/selectors.js]: Fixed

File: src/redux/modules/transactionsList/selectors.js (hunk 5 above)
Comment (dzheki4ak) [reply src/redux/modules/transactionsList/selectors.js]: I have removed these selectors due to I will not use them. Instead I will use a helper function and useSelector hook;

File: src/redux/store.js (hunk 6 above)
Comment (dzheki4ak) [reply src/redux/store.js:2]: Removed from package JSON

File: src/components/BalancePage/BalancePage.jsx (hunk 2 above)
Comment (dzheki4ak) [reply src/components/BalancePage/BalancePage.jsx]: Fixed

File: src/components/BalancePage/BalancePage.jsx (hunk 3 above)
Comment (dzheki4ak) [reply src/components/BalancePage/BalancePage.jsx:20]: Done. Can you please check?

**Comment by vitcool** [comment]
cool, approved!

**Pull request merged**